single_kernel_mongo/state/charm_state.py

```diff
@@ -870,3 +875,21 @@ def mongo_config(self) -> MongoConfiguration:
         return self.mongos_config
 
     # END: Configuration accessors
+
+    def get_secret_from_id(self, secret_id: str) -> dict[str, str]:
+        """Resolve the given id of a Juju secret and return the content as a dict.
+
+        Args:
+            secret_id (str): The id of the secret.
+
+        Returns:
+            dict: The content of the secret.
+        """
+        try:
+            secret_content = self.charm.model.get_secret(id=secret_id).get_content(refresh=True)
+        except SecretNotFoundError:
+            raise SecretNotFoundError(f"The secret '{secret_id}' does not exist.")
+        except ModelError:
+            raise
+
+        return secret_content
```

Comment:
Nit: already map each username to its `MongoDBUser` here and discard all other values in the secret from the map. That way you know this function only returns valid users.